packages/indexer/native/src/lib.rs

```diff
@@ -435,11 +437,25 @@ declare_types! {
             };
             let mut start_key = vec![prefix as u8];
             start_key.extend_from_slice(&script.as_slice()[SCRIPT_SERIALIZE_OFFSET..]);
+            let mut end_key = start_key.clone();
+            let from_block = cx.argument::<JsValue>(3)?;
+            if from_block.is_a::<JsNumber>() {
+                let from_block_number = from_block.downcast::<JsNumber>().or_throw(&mut cx)?.value() as u64;
+                start_key.extend_from_slice(&from_block_number.to_be_bytes());
+            }
+            let to_block = cx.argument::<JsValue>(4)?;
+            if to_block.is_a::<JsNumber>() {
+                // here set to_block_number as toBlock + 1, making the toBlock included in query range.
+                let to_block_number = to_block.downcast::<JsNumber>().or_throw(&mut cx)?.value() as u64 + 1;
+                end_key.extend_from_slice(&to_block_number.to_be_bytes());
+            } else {
+                end_key.extend_from_slice(&u64::MAX.to_be_bytes());
+            }
             let iter = store.iter(&start_key, IteratorDirection::Forward);
             if iter.is_err() {
                 return cx.throw_error("Error creating iterator!");
             }
-            let iter = iter.unwrap().take_while(move |(key, _)| key.starts_with(&start_key));
+            let iter = iter.unwrap().take_while(move |(key, _)| key.to_vec() < end_key);
```

Comment:
What about using `<=` here, so we won't need to add 1 to `to_block_number` above? Although the possibility is extremely rare, adding 1 without bound check is dangerous, it also will generate warnings when we use certain automatic code analysis tools. I doubt this will be a real issue, it is more of a cosmetic fix IMO.

Reply:
I considered using `<=` in the first place, but decided not after checking the `TxLockScript | TxTypeScript` serialization here, https://github.com/quake/ckb-indexer/blob/94b4db3a5e793999d970c838e3843c72ef164e8a/src/indexer.rs#L97. 

There might be a inconsistent situation: 
1. bytes after `block_number` are all-zeros;
2. bytes after `block_number` are not all-zeros;
Both `Tx

Reply:
Ah I see, yes you are right here. In that sense ignore this.